Hints.py

```diff
@@ -1145,19 +1145,22 @@ def get_junk_hint(spoiler: Spoiler, world: World, checked: set[str]) -> HintRetu
 
 def get_important_check_hint(spoiler: Spoiler, world: World, checked: set[str]) -> HintReturn:
     top_level_locations = []
+    empty_dungeons = [dungeon for dungeon in world.precompleted_dungeons if world.precompleted_dungeons[dungeon]]
     for location in world.get_filled_locations():
         if (HintArea.at(location).text(world.settings.clearer_hints) not in top_level_locations
                 and (HintArea.at(location).text(world.settings.clearer_hints) + ' Important Check') not in checked
-                and HintArea.at(location) != HintArea.ROOT):
+                and HintArea.at(location) != HintArea.ROOT
+                and HintArea.at(location).dungeon_name not in empty_dungeons # prevent pre-completed dungeons from being hinted
+                and not location.locked): # prevent areas with unshuffled checks from being hinted
             top_level_locations.append(HintArea.at(location).text(world.settings.clearer_hints))
+    if not top_level_locations:
+        return None
     hint_loc = random.choice(top_level_locations)
     item_count = 0
     for location in world.get_filled_locations():
         region = HintArea.at(location).text(world.settings.clearer_hints)
         if region == hint_loc:
             if (location.item.majoritem
-                # exclude locked items
-                and not location.locked
```

Comment:
I don't think this is redundant? If the hint area has a mix of shuffled and locked locations, it wouldn't be excluded by the check above, so removing this adds the locked locations to the count, which is a functional change.

Reply:
`top_level_locations` is a misnomer, it's actually a list of (human-readable representations of) hint areas. So a locked location could still be iterated over here, if its hint area was added to `top_level_locations` via another shuffled location in the same hint area.